### PR TITLE
Api

### PR DESCRIPTION
**Pull request by Vgonma**
# Add api enpoints for:
- Listing all posts for a user.
- Listing all comments for a user's post.
- Adding a comment to a post. Remember that the owner of the comment is the user that makes it.
- All API endpoints receive JSON and respond JSON as well.

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly ensure to remove Live demo link from the ReadMe if app is not yet deployed.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Vgonma** [comment]
> Hi Team,
> 
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> 
> ## Invalid Code Review Request
> Kindly ensure to remove Live demo link from the ReadMe if app is not yet deployed.
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **does not count** into the code reviews limit._

### We believe this is no reason to invalidate the review since previous code reviewers asked us to add the demo link even with no working link.  Either way, the Readme file was updated and a tag that specifies that the link is not working is now visible next to the link.

**Comment by Whoistolu** [comment]
> > Hi Team,
> > While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> > ## Invalid Code Review Request
> > Kindly ensure to remove Live demo link from the ReadMe if app is not yet deployed.
> > Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> > Cheers and Happy coding!👏👏👏
> > Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> > **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> > _Invalid Code Review Request **does not count** into the code reviews limit._
> 
> ### We believe this is no reason to invalidate the review since previous code reviewers asked us to add the demo link even with no working link. Either way, the Readme file was updated and a tag that specifies that the link is not working is now visible next to the link.

Hello Team,
Sorry about the experience with the CR that told you to add the Live Demo section even though it was not working.
We want to help keep your project documentation as professional as possible, that is why we do recommend that you only add the live demo link after the project is deployed.

Thank you for adhereing 💪🏽 
I will go ahead to perform a review right away.

File: app/controllers/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    comments = Post.find(params[:post_id]).comments
+    render json: comments, status: :ok
+  end
+
+  def create
+    comment = Comment.new(comment_params)
+    if comment.save
+      render json: comment, status: :created
+    else
+      render json: { errors: comment.errors }, status: :unprocessable_entity
```
Comment (Whoistolu) [inline app/controllers/api/v1/comments_controller.rb:14]: - Great job rendering JSON 👍🏽

File: config/routes.rb [hunk 2]
```diff
@@ -6,5 +6,13 @@
...
-  # post '/users/:user_id/posts/:post_id/likes', to likes#create
+  namespace :api do
+    namespace :v1 do
+      resources :users do
+        resources :posts do
+          resources :comments
+        end
+      end
+    end
+  end
```
Comment (Whoistolu) [inline config/routes.rb:17]: - Excellent job is done 👍🏽 
Screenshot 2023-04-14 at 17 11 48

**Review by Whoistolu — APPROVED** [review]
### Hi #2 @Vgonma and @belaymit 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

### Highlight
🔆 All API endpoints receives JSON and responds in a JSON format as well
🔆 Has API endpoint that returns all post in JSON format
🔆 Has API endpoint that returns all comments for a post
🔆 Has API endpoint that adds' a comment to a post

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**